ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionsOperation.java

```diff
@@ -44,11 +60,49 @@ public int execute() throws HiveException {
       throw new HiveException(ErrorMsg.TABLE_NOT_PARTITIONED, desc.getTabName());
     }
 
-    List<String> parts = null;
-    if (desc.getPartSpec() != null) {
-      parts = context.getDb().getPartitionNames(tbl.getDbName(), tbl.getTableName(), desc.getPartSpec(), (short) -1);
+    List<String> parts;
+    if (desc.getCond() != null || desc.getOrder() != null) {
+      ExprNodeDesc pred = desc.getCond();
```

Comment:
This is long, please move it to a method.

Reply:
Move the logic into method getPartNamesByConstraints alone